cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/ValidatorImpl.java

```diff
@@ -0,0 +1,204 @@
+                        " is declared in a file named \"" + fileName + "." + fileExtension.getValue() + "\"" +
+                        ", it should be declared in a file named \"" + executableName + "." + fileExtension.getValue() + "\""));
+            }
+        }
+    }
+
+    @Override
+    public void validateInputNamesDifferentFromOutputNames(ExecutableModellingResult result) {
+        List<Input> inputs = result.getExecutable().getInputs();
+        List<Output> outputs = result.getExecutable().getOutputs();
+        String errorMessage = "Inputs and outputs names should be different for \"" +
+                result.getExecutable().getId() + "\". " +
+                "Please rename input/output \"" + NAME_PLACEHOLDER + "\"";
+        try {
+            validateListsHaveMutuallyExclusiveNames(inputs, outputs, errorMessage);
+        } catch (RuntimeException e) {
+            result.getErrors().add(e);
+        }
+    }
+
+    @Override
+    public List<RuntimeException> validateModelWithDependencies(
+            Executable executable,
+            Map<String, Executable> filteredDependencies) {
+        Map<String, Executable> dependencies = new HashMap<>(filteredDependencies);
+        dependencies.put(executable.getId(), executable);
+        Set<Executable> verifiedExecutables = new HashSet<>();
+        return validateModelWithDependencies(executable, dependencies, verifiedExecutables, new ArrayList<RuntimeException>());
+    }
+
+    private boolean navigationListContainsKey(List<Map<String, String>> stepNavigations, String resultName) {
+        for (Map<String, String> map : stepNavigations) {
+            if (map.containsKey(resultName)) return true;
+        }
+        return false;
+    }
+
+    private List<RuntimeException> validateModelWithDependencies(
+            Executable executable,
+            Map<String, Executable> dependencies,
+            Set<Executable> verifiedExecutables,
+            List<RuntimeException> errors) {
+        //validate that all required & non private parameters with no default value of a reference are provided
+        if(!SlangTextualKeys.FLOW_TYPE.equals(executable.getType()) || verifiedExecutables.contains(executable)){
+            return errors;
+        }
+        verifiedExecutables.add(executable);
+
+        Flow flow = (Flow) executable;
+        Collection<Step> steps = flow.getWorkflow().getSteps();
+        Set<Executable> flowReferences = new HashSet<>();
+
+        for (Step step : steps) {
+            String refId = step.getRefId();
+            Executable reference = dependencies.get(refId);
+            List<String> mandatoryInputNames = getMandatoryInputNames(reference);
+            List<String> stepInputNames = getStepInputNames(step);
+            List<String> inputsNotWired = getInputsNotWired(mandatoryInputNames, stepInputNames);
+
+            try {
+                validateInputNamesEmpty(inputsNotWired, flow, step, reference);
```

Comment:
I think we should rename this method to be ~ validateMandatoryInputsAreWired and move that logic into the moethod:

```java
List<String> mandatoryInputNames = getMandatoryInputNames(reference);
 +            List<String> stepInputNames = getStepInputNames(step);
 +            List<String> inputsNotWired = getInputsNotWired(mandatoryInputNames, stepInputNames);
```